### DISCUSSION TITLE
Enable/Disable Main window buttons based on authorization.

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#1665] [Questions]
I have an enterprise application for which I would like to enable/disable buttons based on authorization and state (is a customer found then enable these buttons, and disable those, etc.)  I am wondering if anyone has any ideas on how to do this?   I have about 100 views that are grouped by tabs (10 different tabs).  I know how to bind the enabled attribute of the button, but I'm trying to stay away from 100 different button properties.  What are your thoughts?

**Comment by rockfordlhotka** [comment]
You can create custom authorization (authz) rules. Each property and data portal operation can have exactly one authz rule attached.

So, for example, you can use a custom authz rule for the save (insert/update/delete) operation set on a root type, and that'll impact `IsSaveable`.

On a per-property basis you can have an authz rule for read and write operations.

And on a per-method basis (arbitrary method on a domain object) you can have an authz rule to control whether the method can be invoked.

All of those can be made bindable, though _how_ that is done varies by UI framework.
